### PR TITLE
improve(create): add web3 connection

### PR DESCRIPTION
**Pull request by Gamaranto**
**Motivation**
This PR adds a `useConnection` hook to interact with different web3 wallets and make it easy to retrieve information through the app. The hook uses `bnc-onboard` to allow users to connect their wallet. 

**Summary**

The PR adds a `useConnection` hook to interact with Ethereum.

**Details**

The UI added is just to test the app manually and is expected to change, so it looks ugly and empty for now. 
For testing, I've decided to avoid making any network requests as I want the test to be fast and work offline, the tradeoff is an easier to read and maintain test for a minimal loss in confidence, which I think is worth it. 

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

Signed-off-by: Francesco Baccetti <francesco@umaproject.org>

### CODE REVIEW

**Review by mrice32 — APPROVED** [review]
LGTM!

**Comment by mrice32** [comment]
Also really cool to see the jest testing in action.

File: src/__test__/useConnection.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+beforeAll(async () => {
+  fakeProvider = new ethers.providers.Web3Provider(mockProviderFn);
+  fakeNetwork = await fakeProvider.getNetwork();
+});
+
+test("should connect to the provider and save the data", async () => {
+  const { promise, resolve } = deferred();
+
+  mockedOnboard.mockImplementation(({ subscriptions }: Initialization) => {
+    promise.then(() => {
```
Comment (daywiss) [inline src/__test__/useConnection.test.ts:48]: we talked about doing this as an event emitter instead. will leave it up to you though if you want to change it.

**Review by daywiss — APPROVED** [review]
looks great

**Review by daywiss — APPROVED** [review]
nice!

**Pull request merged**